doPointSamplingTool.py

```diff
@@ -18,454 +18,626 @@
-   # This message box may to make some confusion, if user press OK while "too long name" is still under edition.
-   # In this case, the message box pops up (correctly) and then the OK button becomes pressed, but the self.accept method is not called.
-   # This pressed button with no action may to look like a hang up.
-   # I've no idea either
-   #  how to pull this button up, or
-   #  how to forse execution the self.accept method, or
-   #  how to don't allow user to exceed 10 chars limit in QTableWidget cell. (THIS OPTION WOULD BE THE BEST!)
-  self.fieldsTable.resizeColumnsToContents()
-
-
-
- def outFile(self): # by Carson Farmer 2008
-  # display file dialog for output file
-  self.outShape.clear()
-  outName, _ = QFileDialog().getSaveFileName(self, self.tr("Output file"), ".",
-                                             self.tr("GeoPackages(*.gpkg);;Comma separated values (*.csv);;Shapefiles (*.shp)"),
-                                             options = QFileDialog.DontConfirmOverwrite)
-  outPath = QFileInfo(outName).absoluteFilePath()
-  if not outPath.upper().endswith('.GPKG') and not outPath.upper().endswith('.CSV') and not outPath.upper().endswith('.SHP'):
-   outPath += '.gpkg'
-  if outName:
-   self.outShape.clear()
-   self.outShape.insert(outPath)
-
-
-
- def accept(self): # Called when "OK" button pressed (based on the Carson Farmer's PointsInPoly Plugin, 2008)
-  # check if all fields are filled up
-  self.statusLabel.setText(self.tr("Check input values, please!"))
-  nothingSelected = True
-  for i in self.polyItems:
-   for j in range(1, len(self.polyItems[i])):
-    if self.polyItems[i][j][2]:
-     nothingSelected = False
-  for i in self.rastItems:
-   for j in range(1, len(self.rastItems[i])):
-    if self.rastItems[i][j][2]:
-     nothingSelected = False
+Ui_Dialog = uic.loadUiType(
+    os.path.join(os.path.dirname(__file__), "pointSamplingToolUi.ui")
+)[0]
 
-  if self.inSample.currentText() == "":
-   self.tabWidget.setCurrentIndex(0)
-   QMessageBox.information(self, self.tr("Point Sampling Tool"), self.tr("Please select vector layer containing the sampling points"))
-   return
-  if nothingSelected:
-   self.tabWidget.setCurrentIndex(0)
-   QMessageBox.information(self, self.tr("Point Sampling Tool"), self.tr("Please select at least one polygon attribute or raster band"))
-   return
-  if self.outShape.text() == "":
-   self.tabWidget.setCurrentIndex(0)
-   QMessageBox.information(self, self.tr("Point Sampling Tool"), self.tr("Please specify output file name"))
-   return
-  # check if destination field names are unique
-  if not self.testFieldsNames(self.fields):
-   self.updateFieldsTable()
-   self.tabWidget.setCurrentIndex(1)
-   QMessageBox.warning(self, self.tr("Point Sampling Tool"), self.tr("At least two field names are the same!\nPlease type unique names."))
-   return
+QT_VERSION_INT = int(QT_VERSION_STR.split(".")[0])
+QT_CHECKED = Qt.Checked if QT_VERSION_INT <= 5 else Qt.CheckState.Checked
```

Comment:
I'm pretty sure the qualified form Qt.CheckState.Checked was already present in Qt5 from beginning or almost beginning, so it looks safe to me to drop the old one.